friends/src/App.js

```diff
@@ -0,0 +1,112 @@
+import React, { Component } from 'react';
+import axios from 'axios';
+import './App.css';
+
+import FriendsList from './components/FriendsList';
+import FriendForm from './components/FriendForm';
+import UpdateFriendForm from './components/UpdateFriendForm';
+
+class App extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      friends: [],
+      friend: {
+        name: '',
+        age: '',
+        email: ''
+      },
+      updating: false
+    };
+  }
+
+  componentDidMount() {
+    axios.get('http://localhost:5000/friends')
+      .then(res => {
+        this.setState({
+          friends: res.data
+        })
+      })
+      .catch(err => console.log(err));
+  }
+
+  handleChanges = e => {
+    this.setState({
+      friend: {
+        ...this.state.friend,
+        [e.target.name]: e.target.value
+      }
+    });
+  }
+
+  addFriend = friend => {
+    axios
+      .post('http://localhost:5000/friends', friend)
+      .then(res => this.setState({
+        friends: res.data.friend
```

Comment:
This is returning 1 friend object and putting it in the friends array on state, which makes it so that the map method breaks in the FriendsList component.

Reply:
Nvm, it returns to the friends array 'undefined'. You could get rid of the .friend and it will put all friends in the array on state.